datajunction-clients/python/datajunction/client.py

```diff
@@ -31,31 +29,76 @@ def list_namespaces(self, prefix: Optional[str] = None) -> List[str]:
 
     def list_dimensions(self, namespace: Optional[str] = None) -> List[str]:
         """
-        List dimension nodes under given namespace or all.
+        List dimension nodes for a given namespace or all.
         """
         if namespace:
             return self._get_nodes_in_namespace(
                 namespace=namespace,
-                type_=models.NodeType.DIMENSION.value,
+                type_=models.NodeType.DIMENSION,
             )
-        return self._session.get("/dimensions/").json()
+        return self._get_all_nodes(type_=models.NodeType.DIMENSION)
 
     def list_metrics(self, namespace: Optional[str] = None) -> List[str]:
         """
-        List metric nodes for given namespace or all.
+        List metric nodes for a given namespace or all.
         """
         if namespace:
             return self._get_nodes_in_namespace(
                 namespace=namespace,
-                type_=models.NodeType.METRIC.value,
+                type_=models.NodeType.METRIC,
             )
-        return self._session.get("/metrics/").json()
+        return self._get_all_nodes(type_=models.NodeType.METRIC)
 
     def list_cubes(self, namespace: Optional[str] = None) -> List[str]:
         """
-        List metric nodes for given namespace or all.
-        TODO
+        List cube nodes for a given namespace or all.
         """
+        if namespace:
+            return self._get_nodes_in_namespace(
+                namespace=namespace,
+                type_=models.NodeType.CUBE,
+            )
+        return self._get_all_nodes(type_=models.NodeType.CUBE)
+
+    #
+    # List other nodes: sources, transforms, all.
+    #
+    def list_sources(self, namespace: Optional[str] = None) -> List[str]:
```

Comment:
Should there also be an endpoint to list source nodes that were created via the register-a-table API endpoint? I know that users can also do that just by calling `list_sources(namespace="source")` (or whatever the default namespace was configured to be server-side), but they might not know what the default registered table sources namespace is. 

No need to do it in this PR, of course!